Mathlib/Order/InitialSeg.lean

```diff
@@ -265,6 +268,21 @@ instance : CoeOut (r ≺i s) (r ↪r s) :=
 instance : CoeFun (r ≺i s) fun _ => α → β :=
   ⟨fun f => f⟩
 
+instance [IsIrrefl β s] [IsTrichotomous β s] : FunLike (r ≺i s) α β where
+  coe f := f.toFun
+  coe_injective' := by
+    rintro ⟨f, a, hf⟩ ⟨g, b, hg⟩ h
+    cases (DFunLike.coe_injective h : f = g)
+    congr
+    refine extensional_of_trichotomous_of_irrefl s fun x ↦ ?_
+    rw [← hf, hg]
```

Comment:
Having both of these at once is a bad idea, since now you have two different spellings for coercion to function.

Reply:
I think it would be better to instead prove that `toRelEmbedding` is injective.